### DISCUSSION TITLE
Welcome to pivpn-tap-web-ui Discussions!

### DISCUSSION THREAD

**Discussion by bnhf** [bnhf/openvpn-admin-plus#2] [Q&A] [answered]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by dlane3221** [comment]
I have followed the thread and have some similarities.  I did adjust the external server address and it can be reached.  The Management Interface Address does not work.  If I change it from 127.0.0.1:2080, the system does not respond.  I can log in, but it then fails when it goes to the home page.

**Comment by bnhf** [comment]
@dlane3221

It sounds like your active server.conf file doesn't contain the required ```management 0.0.0.0 2080``` directive.

Did you do an initial run-through the OpenVPN Config page to get your PiVPN Server configured? It's desirable to do that, so what shows in the Web UI matches what's actually in your server.conf file in the /etc/openvpn directory.

Once the management port is opened, then you should be able change the Management interface address field using the LAN IP of the server followed by a colon and the port # (2080 by default). For example, ```192.168.1.50:2080```

**Comment by bnhf** [comment]
@dlane3221 

Also, feel free to post screenshots of the two configuration pages from the Web UI -- you'd probably want to redact your external server address and port number if possible. Those pages plus the contents of your /etc/openvpn/server.conf file would give me a pretty complete picture. Redacting the port number in server.conf is a good idea here too.

BTW, you can also test whether the OpenVPN management port is open using PuTTY. Parameters are hostname or IP of the server, port number of 2080, and "Raw" as the connection type. If successful, you should see ```>INFO:OpenVPN Management Interface Version 3 -- type 'help' for more info```. Type ```exit``` to break the connection. If you don't get a response like the above, the management port is not configured properly (which is the default state).

**Comment by dlane3221** [comment]
dev tun
proto udp
port 1***
ca /etc/openvpn/easy-rsa/pki/ca.crt
cert /etc/openvpn/easy-rsa/pki/issued/pihole_************************************.crt
key /etc/openvpn/easy-rsa/pki/private/pihole_************************************.key
dh none
ecdh-curve prime256v1
topology subnet
server 10.8.0.0 255.255.255.0
management 192.168.1.254 ****
# Set your primary domain name server address for clients
push "dhcp-option DNS 10.8.0.1"
push "block-outside-dns"
# Override the Client default gateway by using 0.0.0.0/1 and
# 128.0.0.0/1 rather than 0.0.0.0/0. This has the benefit of
# overriding but not wiping out the original default gateway.
push "redirect-gateway def1"
client-to-client
client-config-dir /etc/openvpn/ccd
keepalive 15 120
remote-cert-tls client
tls-version-min 1.2
tls-crypt /etc/openvpn/easy-rsa/pki/ta.key
cipher AES-256-CBC
auth SHA256
user ******
group *******
persist-key
persist-tun
crl-verify /etc/openvpn/crl.pem
status /var/log/openvpn-status.log 20
status-version 3
syslog
verb 3
#DuplicateCNs allow access control on a less-granular, per user basis.
#Remove # if you will manage access by user instead of device. 
#duplicate-cn
# Generated for use by PiVPN.io

I was able to reach the management server using Putty and got the expected results which now confuses me more.

Thanks

David

**Comment by bnhf** [comment]
@dlane3221 

The OpenVPN management directive needs to be set to ```management 0.0.0.0 2080``` so that it listens on all interfaces. The Management interface in the web UI configuration needs to be ```192.168.1.254:2080```, based on what I'm seeing in your server.conf file.

That should do the trick.

**Comment by bnhf** [comment]
@dlane3221 

You'll need to restart OpenVPN or reboot your server after making the above change.

**Comment by dlane3221** [comment]
I will give it a shot but I tried that combination and the pivot management
page logs in then does not respond. I will let you know.

**Comment by dlane3221** [comment]
Set the management option to 0.0.0.0 2080 and set the interface to 192.168.1.254:2080

And this is what I see when I refresh…

**Comment by dlane3221** [comment]
Set the settings back to 127.0.0.1:2080 and the page works again

David

**Comment by bnhf** [comment]
@dlane3221 

Whatever you were meaning to add after "And this is what I see when I refresh…" didn't come through.

**Comment by bnhf** [comment]
@dlane3221 

Give me a few details on your setup please:

What OS are you running, and what version?
Is your Docker installation standard (i.e. not rootless)?
Are you using Portainer, and linking to the GitHub Repo for your pivpn-tap-web-ui stack?
Did you set your environment variables in the Portainer stack per the README?
It appears that you manually edited the server.conf file, which makes me wonder if the container is able to write to /etc/openvpn -- any particular reason for not doing the final configuration through the web ui?

**Comment by dlane3221** [comment]
The OS is – 

PRETTY_NAME="Raspbian GNU/Linux 10 (buster)"
NAME="Raspbian GNU/Linux"
VERSION_ID="10"
VERSION="10 (buster)"
VERSION_CODENAME=buster
ID=raspbian
ID_LIKE=debian
HOME_URL=http://www.raspbian.org/
SUPPORT_URL=http://www.raspbian.org/RaspbianForums
BUG_REPORT_URL=http://www.raspbian.org/RaspbianBugs

I believe the docker install is standard.  I installed via get-docker.sh command.

I followed your instructions for the portainer installation and set the variables as instructed.

I do manually edit the server.conf.  Last night I used the web to change the management interface setting and it created a new sever.conf and blew up my VPN to the point no traffic as moving through.  I found that mistake this morning and noted to never do that again.

**Comment by bnhf** [comment]
@dlane3221 

I'm looking for ramifications of not using the in-built server.conf configurator, as that's what I intended when developing the web ui.  First thing that jumps to my attention is the log file location. The container only has access to /etc/openvpn, so that's where the logs need to be written.

In your server.conf file, comment out the standard PiVPN log locations like so, and add the appropriate directive for the web ui:
```
# status /var/log/openvpn-status.log 20
# syslog
# status-version 3
log /etc/openvpn/openvpn.log
```
You should start to see proper log data after an OpenVPN restart, or server reboot. Also, as I've said, the management directive needs to be set to ```management 0.0.0.0 2080```, so confirm that.

As far as the Management interface setting in the web ui is concerned, the IP of the Ethernet port on the Raspberry Pi (assuming you're using Ethernet of course -- which you should be) followed by a colon and the management port number (2080 is the default) is the only correct entry. Anything else is a misconfiguration.

The only other directives that you're using that I don't see in the generated server.conf are ```user``` and ```group```. Since those two relate to permissions, I'd suggest commenting them out as well to be sure.

**Comment by dlane3221** [comment]
That also broke the web system when those lines were changed.  As I was looking it over and comparing to the your web OpenVPN configuration, I noticed they are not the same.  My config is set to TUN and the web says it is a TAP0.  I also do have the log file in /etc/openvp/ already and it does appear in the web interface.  I also found more settings in /etc/pivpn/openvpn/.  This config gets more detailed about my particular installation (my original install settings).  This file is setupvars.conf and it could be tied to your system, I am not sure.  I also do not find another server.conf on the system.  Why do they not match?

I did try to change the web interface to what I know it should be and that was when sever.conf was created in /etc/openvpn/ and stopped all my VPN traffic and would not allow clients to connect.  That is why/when I started editing the conf manually.

David

**Comment by bnhf** [comment]
@dlane3221 

When you initially launch the OpenVPNAdmin web ui the values that are shown in Configuration - OpenVPN Config and Configuration -Settings are defaults only. The idea is for you to enter your own values, using the defaults as a guide for format, relative file paths and the like. Once you've done that and click Save & Apply, the original server.conf file created by PiVPN is replaced by one created by OpenVPNAdmin, and the values will match. Since you never did that, none of the values are actually set in the database and the defaults are still shown.

I'd really like to figure out what's going on here, but I'm going to need some better data from you to do that. My first request is for you to stop replying by e-mail, as that adds loads of extraneous text to the discussion on GitHub. Please go directly to the Github page for this project, and post your comments in the discussion we have going. I've cleaned up everything you've sent so far, as your responses included all kinds of extra e-mail related header info and my comment repeated back. Also, using GitHub you'll be able to add screenshots of what you're seeing, which would be very helpful.

For now, I'd like to verify your PiVPN setup, so please post the output from ```pivpn -d``` executed from the Raspbian command line.

**Comment by bnhf** [comment]
@dlane3221 

To give you an idea of the screenshots that would be helpful, here are some of mine from a test setup, which is also a Raspberry Pi running Raspbian. In my case it's a an RPi3, with the Debian Bullseye version of Raspbian:

![Screenshot 2022-11-16 081855](https://user-images.githubusercontent.com/41088895/202223406-ada0cbd9-82bf-4cde-baf8-f7b2a721cac6.png)

![Screenshot 2022-11-16 081714](https://user-images.githubusercontent.com/41088895/202223446-6108088c-5619-4c39-bbb9-fe2a2c835edb.png)

![screenshot-raspberrypi6_8080-2022 11 16-08_19_55](https://user-images.githubusercontent.com/41088895/202223484-0a6f7324-b476-43db-a8bf-02d9ff4210e1.png)

My OpenVPN config is a bit different then yours as I'm using a TAP interface, so that my connected clients appear as though they're on the local network -- complete with an IP address assigned by the LAN.

**Comment by dlane3221** [comment]
Thank you for the clarification on the expected behavior.  I assumed (incorrectly) that you read in the server.conf to the web interface.  I have since matched my original config to the web config and the system did write that to the server.conf as it should.  All good as my endpoints still work!  

![image](https://user-images.githubusercontent.com/118311772/202231635-81e1c92e-c51c-417a-8466-fa2d3f902e65.png)

![image](https://user-images.githubusercontent.com/118311772/202233406-4a89e30c-c2f5-4c6f-b8e6-5c0673ac0034.png)

pi@pihole:/etc/openvpn $ pivpn -d
::: Generating Debug Output
:::: PiVPN debug ::::
:::: Latest commit ::::
commit 75acaf97ae4457bbfdf9ec5ea0df5c6380bf5c1a
Author: 4s3ti 4s3ti@protonmail.com
Date: Fri Jan 8 16:04:21 2021 +0100

Change issue template

Changed issue template from pivpn.dev to pivpn.io

=============================================

:::: Installation settings ::::
PLAT=Raspbian
OSCN=buster
USING_UFW=0
IPv4dev=eth0
dhcpReserv=1
IPv4addr=192.168*./24
IPv4gw=192.168..*
install_user=pi
install_home=/home/pi
VPN=openvpn
pivpnPROTO=udp
pivpnPORT=1194
pivpnDNS1=10.8.0.1
pivpnDNS2=
pivpnSEARCHDOMAIN=
pivpnHOST=REDACTED
TWO_POINT_FOUR=1
pivpnENCRYPT=256
USE_PREDEFINED_DH_PARAM=
INPUT_CHAIN_EDITED=0
FORWARD_CHAIN_EDITED=0
pivpnDEV=tun0
pivpnNET=10.8.0.0
subnetClass=24
UNATTUPG=0
INSTALLED_PACKAGES=(iptables-persistent openvpn grepcidr expect)
HELP_SHOWN=1
:::: Server configuration shown below ::::
management 0.0.0.0 2080
dev tun
proto udp
port 1194

ca easy-rsa/pki/ca.crt
cert easy-rsa/pki/issued/pihole_652df33c-579b-4ec8-b874-4cfbb94824a9.crt
key easy-rsa/pki/private/pihole_652df33c-579b-4ec8-b874-4cfbb94824a9.key
dh none
ecdh-curve prime256v1

topology subnet
server 10.8.0.0 255.255.255.0
ifconfig-pool-persist ipp.txt
push "dhcp-option DNS 8.8.8.8"
push "dhcp-option DNS 8.8.4.4"

keepalive 15 120
remote-cert-tls client
tls-version-min 1.2
tls-crypt easy-rsa/pki/ta.key
cipher AES-256-GCM
auth SHA256

persist-key
persist-tun
crl-verify /etc/openvpn/crl.pem

status /etc/openvpn/openvpn-status.log 20
status-version 3
syslog
log /etc/openvpn/openvpn.log
verb 3
mute 10

Set your primary domain name server address for clients
push "dhcp-option DNS 10.8.0.1"
push "block-outside-dns"

Override the Client default gateway by using 0.0.0.0/1 and
128.0.0.0/1 rather than 0.0.0.0/0. This has the benefit of
overriding but not wiping out the original default gateway.
push "redirect-gateway def1"
client-to-client
client-config-dir /etc/openvpn/ccd
:::: Client template file shown below ::::
client
dev tun
proto udp
remote REDACTED 1194
resolv-retry infinite
nobind
remote-cert-tls server
tls-version-min 1.2
verify-x509-name pihole_652df33c-579b-4ec8-b874-4cfbb94824a9 name
cipher AES-256-CBC
auth SHA256
auth-nocache
verb 3
:::: Recursive list of files in ::::
::: /etc/openvpn/easy-rsa/pki shows below :::
/etc/openvpn/easy-rsa/pki/:
BCL-iPhone.ovpn
BCL-Router.ovpn
BCL-Surface.ovpn
ca.crt
crl.pem
Default.txt
DJL_iPhone_New.ovpn
DJL-iPhone.ovpn
DJL-Router.ovpn
DJL-Surface.ovpn
ecparams
index.txt
index.txt.attr
index.txt.attr.old
index.txt.old
issued
MJL-iPhone.ovpn
MJL-Surface.ovpn
MRL-Apt.ovpn
MRL-iPhone.ovpn
MRL-nopass.ovpn
MRL-Surface.ovpn
openssl-easyrsa.cnf
private
renewed
revoked
safessl-easyrsa.cnf
serial
serial.old
SJL-Computer.ovpn
SJL-iPhone.ovpn
ta.key

/etc/openvpn/easy-rsa/pki/ecparams:
prime256v1.pem

/etc/openvpn/easy-rsa/pki/issued:
BCL-iPhone.crt
BCL-Router.crt
BCL-Surface.crt
DJL-iPhone.crt
DJL_iPhone_New.crt
DJL-Router.crt
DJL-Surface.crt
MJL-iPhone.crt
MJL-Surface.crt
MRL-Apt.crt
MRL-iPhone.crt
MRL-nopass.crt
MRL-Surface.crt
pihole_652df33c-579b-4ec8-b874-4cfbb94824a9.crt
SJL-Computer.crt
SJL-iPhone.crt

/etc/openvpn/easy-rsa/pki/private:
BCL-iPhone.key
BCL-Router.key
BCL-Surface.key
ca.key
DJL-iPhone.key
DJL_iPhone_New.key
DJL-Router.key
DJL-Surface.key
MJL-iPhone.key
MJL-Surface.key
MRL-Apt.key
MRL-iPhone.key
MRL-nopass.key
MRL-Surface.key
pihole_652df33c-579b-4ec8-b874-4cfbb94824a9.key
SJL-Computer.key
SJL-iPhone.key

/etc/openvpn/easy-rsa/pki/renewed:
private_by_serial
reqs_by_serial

/etc/openvpn/easy-rsa/pki/renewed/private_by_serial:

/etc/openvpn/easy-rsa/pki/renewed/reqs_by_serial:

/etc/openvpn/easy-rsa/pki/revoked:
private_by_serial
reqs_by_serial

/etc/openvpn/easy-rsa/pki/revoked/private_by_serial:

/etc/openvpn/easy-rsa/pki/revoked/reqs_by_serial:
:::: Self check ::::
:: [OK] IP forwarding is enabled
:: [OK] Iptables MASQUERADE rule set
:: [OK] OpenVPN is running
:: [OK] OpenVPN is enabled (it will automatically start on reboot)
:: [OK] OpenVPN is listening on port 1194/udp
:::: Having trouble connecting? Take a look at the FAQ:
:::: https://github.com/pivpn/pivpn/wiki/FAQ
:::: Snippet of the server log ::::
Nov 16 10:45:57 pihole ovpn-server[8503]: REDACTED:1762 peer info: IV_SSL=OpenSSL_1.1.1q__5_Jul_2022
Nov 16 10:45:57 pihole ovpn-server[8503]: REDACTED:1762 peer info: IV_PLAT_VER=16.1
Nov 16 10:45:57 pihole ovpn-server[8503]: REDACTED:1762 WARNING: 'link-mtu' is present in local config but missing in remote config, local='link-mtu 1569'
Nov 16 10:45:57 pihole ovpn-server[8503]: REDACTED:1762 WARNING: 'tun-mtu' is present in local config but missing in remote config, local='tun-mtu 1500'
Nov 16 10:45:57 pihole ovpn-server[8503]: REDACTED:1762 WARNING: 'tls-crypt' is present in remote config but missing in local config, remote='tls-crypt'
Nov 16 10:45:57 pihole ovpn-server[8503]: REDACTED:1762 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, 256 bit EC, curve: prime256v1
Nov 16 10:45:57 pihole ovpn-server[8503]: REDACTED:1762 [DJL-iPhone] Peer Connection Initiated with [AF_INET]REDACTED:1762
Nov 16 10:45:57 pihole ovpn-server[8503]: DJL-iPhone/REDACTED:1762 OPTIONS IMPORT: reading client specific options from: /etc/openvpn/ccd/DJL-iPhone
Nov 16 10:45:57 pihole ovpn-server[8503]: DJL-iPhone/REDACTED:1762 MULTI: Learn: 10.8.0.2 -> DJL-iPhone/REDACTED:1762
Nov 16 10:45:57 pihole ovpn-server[8503]: DJL-iPhone/REDACTED:1762 MULTI: primary virtual IP for DJL-iPhone/REDACTED:1762: 10.8.0.2
Nov 16 10:45:57 pihole ovpn-server[8503]: DJL-iPhone/REDACTED:1762 PUSH: Received control message: 'PUSH_REQUEST'
Nov 16 10:45:57 pihole ovpn-server[8503]: DJL-iPhone/REDACTED:1762 SENT CONTROL [DJL-iPhone]: 'PUSH_REPLY,dhcp-option DNS 10.8.0.1,block-outside-dns,redirect-gateway def1,route-gateway 10.8.0.1,topology subnet,ping 15,ping-restart 120,ifconfig 10.8.0.2 255.255.255.0,peer-id 2,cipher AES-256-GCM' (status=1)
Nov 16 10:45:57 pihole ovpn-server[8503]: DJL-iPhone/REDACTED:1762 Data Channel: using negotiated cipher 'AES-256-GCM'
Nov 16 10:45:57 pihole ovpn-server[8503]: DJL-iPhone/REDACTED:1762 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Nov 16 10:45:57 pihole ovpn-server[8503]: DJL-iPhone/REDACTED:1762 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
Nov 16 10:46:37 pihole ovpn-server[8503]: event_wait : Interrupted system call (code=4)
Nov 16 10:46:37 pihole ovpn-server[8503]: Closing TUN/TAP interface
Nov 16 10:46:37 pihole ovpn-server[8503]: /sbin/ip addr del dev tun0 10.8.0.1/24
Nov 16 10:46:37 pihole ovpn-server[8503]: Linux ip addr del failed: external program exited with error status: 2
Nov 16 10:46:37 pihole ovpn-server[8503]: SIGTERM[hard,] received, process exiting
:::: Debug complete ::::
:::
::: Debug output completed above.
::: Copy saved to /tmp/debug.log
:::
pi@pihole:/etc/openvpn $

**Comment by dlane3221** [comment]
I still have the issue of changing the management site to my IP.  When I do that, I have an unresponsive web page.  Put it back to 0.0.0.0 or 127.0.01 and the page responds but shows nothing of use.

**Comment by bnhf** [comment]
@dlane3221 

> I still have the issue of changing the management site to my IP. When I do that, I have an unresponsive web page. Put it back to 0.0.0.0 or 127.0.01 and the page responds but shows nothing of use.

And this is the issue. Normally, when the "Management interface address" is set incorrectly, you'll just see a message on the "Home" page saying that you need to fix your configuration. I've never seen, nor has anyone else reported to me, that changing this to the LAN IP address of the OpenVPN server has caused the web ui to become unresponsive.

So, the diagnosis continues. :-) Next thing I'd like you to do is enter ```hostname -I``` (that's a capital i after the dash btw) from the Raspbian command line and post the results here. Don't worry that this shows the internal IP addresses assigned to your RPi -- those addresses are of no use to anyone (except to me trying to sort out where things are going wrong).

**Reply by dlane3221** [reply]
hostname -I
192.168.1.254 172.17.0.1 172.21.0.1 169.254.236.36 10.8.0.1 169.254.106.27 fdc7:79c8:3e6b:4216:73ca:205b:7bb0:90e9
pi@pihole:/etc/openvpn $

**Comment by bnhf** [comment]
@dlane3221 

> hostname -I
> 192.168.1.254 172.17.0.1 172.21.0.1 169.254.236.36 10.8.0.1 169.254.106.27 fdc7:79c8:3e6b:4216:73ca:205b:7bb0:90e9
> pi@pihole:/etc/openvpn $

This output looks as expected except for the two self-assigned addresses that start with 169.254. I have several systems setup with OpenVPN server and OpenVPNAdmin and none of them have any self assigned addresses.

Output from my setups look similar to this -- when I change my OpenVPN server settings to closely match your TUN directives, rather than my usual TAP:
```
pi@raspberrypi6:~ $ hostname -I
192.168.110.102 10.8.0.1 172.19.0.1 172.18.0.1 172.17.0.1 172.21.0.1 172.20.0.1 fdd1:ab5a:e5b5:782:ba27:ebff:fe12:4a81 
```

However I just noticed the hostname shown on your login of "pi@pihole". I take it this RPi is your DNS server as well running PiVPN? I haven't done any testing with PiVPN, OpenVPNAdmin and Pi-hole on the same RPi -- so this would be worth looking into.

Try changing the "Management interface address" to ```pihole:2080```. I used to always use the hostname in this field, but found it sometimes didn't resolve, so I switched to IP. However, if this is your DNS server too, it seems like name resolution should work.

Also, please send me a screenshot of your Portainer - Container page which should look something like this:

![Screenshot 2022-11-16 130708](https://user-images.githubusercontent.com/41088895/202283403-127e1e00-4fce-4d23-9e56-66332d752d34.png)

**Comment by dlane3221** [comment]
![image](https://user-images.githubusercontent.com/118311772/202284518-d2bc8512-7bd4-460b-8428-b889320b6e0d.png)


using pihole or pivpn did not work either and it did not break it either

**Comment by bnhf** [comment]
@dlane3221 

Is Pi-hole your DHCP server too?

Also, could you post the results of ```ip addr```, ```cat /etc/hostname``` and ```cat /etc/hosts``` from the command line?

We'll need to start looking at firewall rules next, as Pi-hole adds a bunch of them.

**Comment by dlane3221** [comment]
pi@pihole:/etc/openvpn $ ip addr
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UP group default qlen 1000
    link/ether b8:27:eb:b9:f4:d0 brd ff:ff:ff:ff:ff:ff
    inet 192.168.1.254/24 brd 192.168.1.255 scope global noprefixroute eth0
       valid_lft forever preferred_lft forever
    inet6 fdc7:79c8:3e6b:4216:73ca:205b:7bb0:90e9/64 scope global dynamic mngtmpaddr noprefixroute
       valid_lft 1663sec preferred_lft 1663sec
    inet6 fe80::95:6d78:ef33:b573/64 scope link
       valid_lft forever preferred_lft forever
3: wlan0: <BROADCAST,MULTICAST> mtu 1500 qdisc noop state DOWN group default qlen 1000
    link/ether b8:27:eb:ec:a1:85 brd ff:ff:ff:ff:ff:ff
5: docker0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default
    link/ether 02:42:9f:4d:31:db brd ff:ff:ff:ff:ff:ff
    inet 172.17.0.1/16 brd 172.17.255.255 scope global docker0
       valid_lft forever preferred_lft forever
    inet6 fe80::42:9fff:fe4d:31db/64 scope link
       valid_lft forever preferred_lft forever
6: br-34b9fcc9bdd9: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default
    link/ether 02:42:1a:b7:6a:4f brd ff:ff:ff:ff:ff:ff
    inet 172.21.0.1/16 brd 172.21.255.255 scope global br-34b9fcc9bdd9
       valid_lft forever preferred_lft forever
    inet6 fe80::42:1aff:feb7:6a4f/64 scope link
       valid_lft forever preferred_lft forever
8: vethd54a49c@if7: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master docker0 state UP group default
    link/ether 6a:53:7c:f4:83:a8 brd ff:ff:ff:ff:ff:ff link-netnsid 0
    inet 169.254.236.36/16 brd 169.254.255.255 scope global noprefixroute vethd54a49c
       valid_lft forever preferred_lft forever
    inet6 fe80::34b4:3435:99ba:e6b/64 scope link
       valid_lft forever preferred_lft forever
13: tun0: <POINTOPOINT,MULTICAST,NOARP,UP,LOWER_UP> mtu 1500 qdisc pfifo_fast state UNKNOWN group default qlen 100
    link/none
    inet 10.8.0.1/24 brd 10.8.0.255 scope global tun0
       valid_lft forever preferred_lft forever
    inet6 fe80::e2af:8ce4:4666:6d18/64 scope link stable-privacy
       valid_lft forever preferred_lft forever
15: veth38c1f65@if14: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue master br-34b9fcc9bdd9 state UP group default
    link/ether de:49:28:8a:57:87 brd ff:ff:ff:ff:ff:ff link-netnsid 1
    inet 169.254.106.27/16 brd 169.254.255.255 scope global noprefixroute veth38c1f65
       valid_lft forever preferred_lft forever
    inet6 fe80::b5a9:4dbc:ad34:dc6/64 scope link
       valid_lft forever preferred_lft forever
pi@pihole:/etc/openvpn $ cat /etc/hostname
pihole
pi@pihole:/etc/openvpn $ cat /etc/hosts
127.0.0.1       localhost
::1             localhost ip6-localhost ip6-loopback
ff02::1         ip6-allnodes
ff02::2         ip6-allrouters

127.0.1.1       pihole

**Comment by dlane3221** [comment]
PiHole is not DHCP.

**Comment by dlane3221** [comment]
pi@pihole:/etc/openvpn $ sudo iptables -L
Chain INPUT (policy ACCEPT)
target     prot opt source               destination

Chain FORWARD (policy ACCEPT)
target     prot opt source               destination
DOCKER-USER  all  --  anywhere             anywhere
DOCKER-ISOLATION-STAGE-1  all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere             ctstate RELATED,ESTABLISHED
DOCKER     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere
ACCEPT     all  --  anywhere             anywhere

Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination

Chain DOCKER (2 references)
target     prot opt source               destination
ACCEPT     tcp  --  anywhere             172.17.0.2           tcp dpt:9000
ACCEPT     tcp  --  anywhere             172.21.0.2           tcp dpt:http-alt

Chain DOCKER-ISOLATION-STAGE-1 (1 references)
target     prot opt source               destination
DOCKER-ISOLATION-STAGE-2  all  --  anywhere             anywhere
DOCKER-ISOLATION-STAGE-2  all  --  anywhere             anywhere
RETURN     all  --  anywhere             anywhere

Chain DOCKER-ISOLATION-STAGE-2 (2 references)
target     prot opt source               destination
DROP       all  --  anywhere             anywhere
DROP       all  --  anywhere             anywhere
RETURN     all  --  anywhere             anywhere

Chain DOCKER-USER (1 references)
target     prot opt source               destination
RETURN     all  --  anywhere             anywhere

**Comment by bnhf** [comment]
@dlane3221 

Good morning David. I'm going to setup a Raspberry Pi that's as close to yours as possible -- to see if I encounter this same issue. So, a couple of questions:

What model RPi do you have (2, 3B, 3B+, 4B)?

Do have any other major software packages installed besides the ones I already know about?

**Comment by dlane3221** [comment]
I am running a 3B for pihole and pivpn.  The pihole does not do DHCP but does DNS.  DHCP is handed out from the router (due to numerous previous reservations).  The wireless is disabled and forced to use wired port.

**Comment by bnhf** [comment]
@dlane3221 

Since I use a pair of Raspberry Pi's as Pi-hole servers myself, I decided to install PiVPN on one of them. It's an RPi 4B, but it has the same Debian 10 32bit version of Raspbian installed on it that you have.

I used the Docker convenience script, installed Portainer, and created a stack for my OpenVPNAdmin interface using the GitHub method. I updated the OpenVPN config using settings that mimic yours, and then rebooted to activate the management interface. Finally, I went into Configuration - Settings and updated the Management interface address and Sever Address (external). And -- everything is working fine:

![Screenshot 2022-11-17 112318](https://user-images.githubusercontent.com/41088895/202527244-9e0009f5-9188-4c79-a1e8-90361dafee78.png)

So now we know that whatever is going on, it's specific to your system. I'm very interested in seeing you up-and-running though, so I'd be happy to build you an SD card with latest and greatest of the OS and the apps you're using and mail it to you. My recommendation would be for this pre-built card to have the current 64bit version of the Raspberry Pi OS, along with the latest Pi-hole, PiVPN and of course my interface.

On one of my two Pi-hole systems, I've actually gone to a Docker-based version of Pi-hole. I also have Organizr, Cockpit (with Navigator), and Watchtower (setup to do container updates on demand) installed. All of their web interfaces are accessible through a single browser tab like so:

![Screenshot 2022-11-17 113418](https://user-images.githubusercontent.com/41088895/202529818-73ccdd05-2097-40fd-84f2-c6307473cffd.png)

So, rather than duplicating your current setup, how about if I do it that way? All the same functionality, but with some extra convenience.  I've done so many similar setups that it's a pretty quick process for me, and I'm happy to do it. If you want me to go ahead, respond back here to let me know and tell me what country you're in. Probably best to send me your address by e-mail though.

**Comment by dlane3221** [comment]
Thank you, but setting up users with a new certificate is painful to me.  I thought it would be nice to have a web interface, but not enough to go through the setups again.  Can you point me to where your application resides to review?  I am thinking there is a routing or permission issue.  I can query the openvpn management port and query the engine with a result and that is the same user that runs the web interface in Portainer.  Are there any additional settings in Docker/Portainer that I have overlooked?

**Comment by bnhf** [comment]
@dlane3221 

You wouldn't actually need to setup a new certificate authority, you could move your keys, certs and users from your existing system. I understand though, if it's not worth it to you.

One thing I did think of, which you may not have done, is to add the "pi" user to the "docker" group. If you didn't do that as part of your Docker install, here's the command:
```
sudo usermod -aG docker pi
```
Reboot after that, and give the Configuration - Settings another shot.

**Comment by dlane3221** [comment]
I removed all Docker and started from scratch.  I did add pi to teh docker group thnking it would be the magic that made it work.  It did not.  When I log into the managment console, I do see entries for the managment client.  The odd thing is the address does not appear.

1668731431,,MANAGEMENT: Client connected from [AF_INET][undef]:2080
1668731431,D,MANAGEMENT: CMD 'status 2'
1668731431,,MANAGEMENT: Client disconnected
1668731755,,MANAGEMENT: Client connected from [AF_INET][undef]:2080
1668731755,D,MANAGEMENT: CMD 'status 2'
1668731755,,MANAGEMENT: Client disconnected
1668731757,,MANAGEMENT: Client connected from [AF_INET][undef]:2080
1668731757,D,MANAGEMENT: CMD 'status 2'
1668731757,,MANAGEMENT: Client disconnected
1668731758,,MANAGEMENT: Client connected from [AF_INET][undef]:2080
1668731758,D,MANAGEMENT: CMD 'status 2'
1668731758,,MANAGEMENT: Client disconnected
1668732471,,MANAGEMENT: Client connected from [AF_INET][undef]:2080
1668732471,D,MANAGEMENT: CMD 'status 2'
1668732471,,MANAGEMENT: Client disconnected
1668732473,,MANAGEMENT: Client connected from [AF_INET][undef]:2080
1668732473,D,MANAGEMENT: CMD 'status 2'
1668732473,,MANAGEMENT: Client disconnected

All other VPN traffic shows an IP.  How does the web quesry the console for updates?  Should there be an antry for the IP as it does for VPN connectinos?  How does the web interfaces with the console?

**Comment by dlane3221** [comment]
I also found this in the logs

 1668728885,,96.44.23.146:47892 VERIFY OK: depth=1, CN=ChangeMe
1668728885,,96.44.23.146:47892 VERIFY KU OK
1668728885,,96.44.23.146:47892 Validating certificate extended key usage
1668728885,,96.44.23.146:47892 ++ Certificate has EKU (str) TLS Web Client Authe     ntication, expects TLS Web Client Authentication
1668728885,,96.44.23.146:47892 VERIFY EKU OK

**Comment by bnhf** [comment, marked as answer]
@dlane3221 

If you're still interested, I believe the issue you were experiencing with the OpenVPNAdmin interface is identical to #4.

The short version of the story is that OpenVPN changed the way some data is presented in the management interface between version 2.4.x and 2.5.x. Older versions of Debian repositories top-out at 2.4.x -- which is not compatible with this updated version of OpenVPNAdmin.

The user referenced in the issue above has updated OpenVPN on his Debian 10 installation, and it's now functioning as intended. Upgrading to Debian 11 (Bullseye or its derivatives) would also be a solution, as its repositories contain version 2.5.x.

**Comment by dlane3221** [comment]
Scott, 

It took a while and required an entire Debian upgrade as OpenVPN would not upgrade alone, but it finally is working.  I did a full upgrade to bullseye from buster and now OpenVPN is 2.5.1 and the dashboard is working!

Thanks

David

**Comment by dlane3221** [comment]
One last question, now that it is all working…

Is there a setting to auto refresh the web page?

Thanks

David

**Comment by bnhf** [comment]
@dlane3221 

There's no setting to auto refresh the "home" page, but I like the idea -- so, I'll see what I can do.

**Comment by karabelnikov** [comment]
Scott, I would like to make some edits to the project on the names, html and css code. Do you mind if they commit and offer it to you for approval? I want to make this project better and more functional, do I really like it?
